server/src/test/java/org/elasticsearch/index/mapper/IgnoredSourceFieldMapperTests.java

```diff
@@ -303,6 +320,45 @@ public void testDisabledSubObject() throws IOException {
             {"boolean_value":%s,"path":{"int_value":%s,"to":{"name":"%s"}}}""", booleanValue, intValue, name), syntheticSource);
     }
 
+    public void testDisabledSubobjectContainsArray() throws IOException {
+        DocumentMapper documentMapper = createMapperService(syntheticSourceMapping(b -> {
+            b.startObject("boolean_value").field("type", "boolean").endObject();
+            b.startObject("path");
+            {
+                b.field("type", "object");
+                b.startObject("properties");
+                {
+                    b.startObject("int_value").field("type", "integer").endObject();
+                    b.startObject("to").field("type", "object").field("enabled", false).endObject();
+                }
+                b.endObject();
+            }
+            b.endObject();
+        })).documentMapper();
+
+        boolean booleanValue = randomBoolean();
+        int intValue = randomInt();
+        var syntheticSource = syntheticSource(documentMapper, b -> {
```

Comment:
Can we also assert the Lucene document that get created here? I think it can be useful to assert the actual `IndexableField` instances for `_ignored_source` field. Maybe the `syntheticSource(...)` method can optionally accept a `Consumer<LuceneDocument>` parameter, that allows to the test to assert the field instances?

Reply:
Hm not sure what would be the benefit, the contents are encoded. The decoded version should show up in the printed source that we already assert on.